src/number_theory/pythagorean_triples.lean

```diff
@@ -59,3 +59,432 @@ def circle_equiv_gen (hk : ∀ x : K, 1 + x^2 ≠ 0) :
 @[simp] lemma circle_equiv_inv (hk : ∀ x : K, 1 + x^2 ≠ 0)
   (v : {p : K × K // p.1^2 + p.2^2 = 1 ∧ p.2 ≠ -1}) :
   (circle_equiv_gen hk).symm v = v.val.1 / (v.val.2 + 1) := by refl
+
+private lemma int.prime.dvd_mul {i j : ℤ} {p : ℕ} (h : nat.prime p) (h2 : (p : ℤ) ∣ i * j)
```

Comment:
A lot of lemmas like this should actually move to other files... so that future work will find them in the "expected" place.

Reply:
This one I agree with, but the ones about gcd (m^2+n^2) (m^2 - n^2)?

Reply:
I would keep a lemma in this file if
* it mentions `pythagorean_triple` or `circle_equiv_gen` ; or
* it seems like a lemma that seems like it will never be useful outside this file

So I would move (at least):
* `parity_eq_abs_parity` & `parity_eq_abs_parity_iff` (with 2 generalized)
* `non_zero_of_coprime` 
* `nat.dvd_of_dvd_pow_two` / `int.dvd_of_dvd_pow_two` (except the first is a direct consequence of `nat.pow_dvd_pow_iff`)

Reply:
Also note that a lot of these names don't follow the naming conventions. Something like `non_zero_of_coprime` should be `ne_zero_of_gcd`. Also, `sqr` is not used for describing `x * x` in the rest of mathlib. `x * x` would be `mul_self` and `x ^ 2` would be `pow_two`.
I understand that this means some names will be longer, but consistency and predictability are really important features for names in a big library.

Reply:
I think I've addressed these now (and did a bit of refactoring)